roles/kubernetes/preinstall/tasks/0030-pre_upgrade.yml

```diff
@@ -23,3 +23,25 @@
   when:
     - old_credential_dir.stat.exists
     - not new_credential_dir.stat.exists
+
+- name: "Pre-upgrade | check if old certificate dir exists"
+  stat:
+    path: "{{ kube_config_dir }}/ssl"
+  register: old_certificate_dir
+  become: yes
+
+- name: "Pre-upgrade | check if new certificate dir exists"
+  stat:
+    path: "{{ kube_cert_dir }}"
+  register: new_certificate_dir
+  become: yes
+  when: old_certificate_dir.stat.exists
+
+- name: "Pre-upgrade | move data from old certificate dir to new"
+  command: mv {{ kube_config_dir }}/ssl {{ kube_cert_dir }}
```

Comment:
Won't this result in `/etc/kubernetes/pki/ssl/ca.crt` instead of `/etc/kubernetes/pki/ca.crt` ?

Reply:
@Miouge1 
No it won't result as you wrote. Cause before moving we are checking that `/etc/kubernetes/pki` not exists.